### PR TITLE
Add new stop_time in trip_update

### PR DESCRIPTION
**Pull request by mbounabi**

### CODE REVIEW

File: kirin/cots/model_maker.py [hunk 1]
```diff
@@ -337,17 +335,40 @@ def _make_trip_update(self, vj, json_train):
...
         """
-        return get_navitia_stop_time_sncf(cr=get_value(pdp, 'cr'),
-                                          ci=get_value(pdp, 'ci'),
-                                          ch=get_value(pdp, 'ch'),
-                                          nav_vj=nav_vj)
+        nav_st, log_dict = get_navitia_stop_time_sncf(cr=get_value(pdp, 'cr'),
+                                                      ci=get_value(pdp, 'ci'),
+                                                      ch=get_value(pdp, 'ch'),
+                                                      nav_vj=nav_vj)
+        if not nav_st:
```
Comment (pbougue) [inline kirin/cots/model_maker.py]: minor: Same here on None vs. cast

File: kirin/cots/model_maker.py [hunk 2]
```diff
@@ -337,17 +335,40 @@ def _make_trip_update(self, vj, json_train):
...
+            nav_stop, log_dict = self.request_navitia_stop_point(cr=get_value(pdp, 'cr'),
+                                                                 ci=get_value(pdp, 'ci'),
+                                                                 ch=get_value(pdp, 'ch'))
+        else:
+            nav_stop = nav_st.get('stop_point', {})
```
Comment (pbougue) [inline kirin/cots/model_maker.py]: A regular `get()` defaulting to `None` would be more consistent with request_navitia_stop_point returning None if no SP is found.

File: kirin/cots/model_maker.py [hunk 3]
```diff
@@ -253,21 +253,20 @@ def _make_trip_update(self, vj, json_train):
 
         # all other status is considered an 'update' of the trip
         trip_update.status = 'update'
-        pdps = _retrieve_interesting_pdp(get_value(json_train, 'listePointDeParcours'))
 
+        pdps = _retrieve_interesting_pdp(get_value(json_train, 'listePointDeParcours'))
         # manage realtime information stop_time by stop_time
         for pdp in pdps:
             # retrieve navitia's stop_time information corresponding to the current COTS pdp
```
Comment (nonifier) [inline kirin/cots/model_maker.py]: The comment needs to be updated as well: "retrieve navitia's ~~stop_time~~ stop_point"

File: kirin/cots/model_maker.py [hunk 4]
```diff
@@ -253,21 +253,20 @@ def _make_trip_update(self, vj, json_train):
...
-            nav_st, log_dict = self._get_navitia_stop_time(pdp, vj.navitia_vj)
+            nav_stop, log_dict = self._get_navitia_stop_point(pdp, vj.navitia_vj)
             if log_dict:
                 record_internal_failure(log_dict['log'], contributor=self.contributor)
                 log_dict.update({'contributor': self.contributor})
                 logging.getLogger(__name__).info('metrology', extra=log_dict)
 
-            if nav_st is None:
+            if not nav_stop:
```
[the remark this answers is not in the document]
Comment (kadhikari) [reply kirin/cots/model_maker.py]: I prefer not instead of None as not will also omit None, '', and 0

**Comment by mbounabi** [comment]
@kadhikari See PR https://github.com/CanalTP/kirin/pull/204
The test is present in it

**Pull request closed without merging**